test/gc/python/test_compiler.py

```diff
@@ -0,0 +1,36 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+# RUN: %python %s | FileCheck %s
+
+from compileall import compile_dir
+
+from gc_mlir.graph_compiler import GraphCompiler
+from gc_mlir.ir import *
+
+
+def run(f):
+    print("\nTEST:", f.__name__)
+    f()
+    return f
+
+
+# CHECK-LABEL: TEST: testCompiler
+@run
+def testCompiler():
+    with Context():
+        module = Module.parse(
+            """
+            func.func @matmul(%arg0: tensor<128x512xf32>, %arg1: tensor<512x256xf32>) -> tensor<128x256xf32> {
+                %0 = onednn_graph.matmul %arg0, %arg1 : (tensor<128x512xf32>, tensor<512x256xf32>) -> tensor<128x256xf32>
+                return %0 : tensor<128x256xf32>
+            }
+            """
+        )
+
+        compiler = GraphCompiler(
+            pipeline="builtin.module(convert-onednn-graph-to-linalg)"
+        )
+        compiler.compile(module)
+        #  CHECK-NOT: onednn_graph.matmul
+        print(module)
```

Comment:
> No need to print the module in the test, but adding some other check with expected result instead?

The ` print(module)` result was used for the FileCheck, for example here the output was used for `#  CHECK-NOT: onednn_graph.matmul`

Reply:
> I don't get why we would run lits through python bindings. Why can't we cover such tests with regular lits?

The porpuse here is to test if the gc dialects and passes are bound correctly, similar to the test dir `llvm-project/mlir/test/python/*.py` in MLIR.

Reply:
Shall we just use python's own unittest modules to do binding tests, if the cases are really about python itself, instead of the text of MLIR.
Please note that `lit` and `FileCheck` are two independent tools from LLVM. I agree that we use `lit` to drive all the tests for both `*.mlir` and `*.py`. However, we don't necessarily need to use FileCheck in this case. `lit` just call this script and check the exit code of the script.
You are free to use FileCheck or python's own unittest for python checkings. You can choose the most fit approach for your case.

Reply:
Sorry, I did not notice the folder of this `test_compiler.py`, and now it was removed, and the test was added into `smoketest.py`.
For the FileCheck part, the python tests under `graph-compiler/test/mlir/test/gc/python/`  using it as the way in the MLIR `llvm-project/mlir/test/python/`, if we have dialects to upstream in the future, the python API tests are also necessary, so we would better using the same check.
Lastly, for the unittest modules  to do binding tests, I got your point, but I think for now it's fine as a part of `gc-check`, just like the python test in MLIR is also a part of `chek-mlir`,